### ISSUE TITLE
Add URLPattern tests for Speculative Loading

### ISSUE THREAD

**Issue by westonruter** [WordPress/performance#1144]
## Feature Description

As @tunetheweb noted in https://github.com/WordPress/performance/pull/1143#issuecomment-2057925186, it would be useful if we could check the validity of the `URLPattern` strings that we're constructing in the Speculative Loading plugin to ensure they are valid.

**Comment by swissspidy** [comment]
On that note, I was thinking about writing a JSON schema for speculation rules to ensure validity of that part too

**Comment by westonruter** [comment]
Yeah, that'll be important especially if the entire rules array is filterable, as the schema could be checked even at runtime.

**Comment by swissspidy** [comment]
Meanwhile, has this been covered by existing tests, e.g. in the core PR https://github.com/WordPress/wordpress-develop/pull/7860/files#diff-6577e99c0aa1f0d10c12d35bc4e071b7402cf833e0a65f6311b8fa505aac4012? It seems like it, so I am inclined to close this as complete.

**Comment by felixarntz** [comment]
+1, this can be closed.
